include/translatable-object.php

```diff
@@ -75,12 +93,19 @@ abstract class PLL_Translatable_Object {
 	 * Constructor.
 	 *
 	 * @since 3.4
+	 * @since 3.7 Changed method's signature.
+	 *
+	 * @param Languages_Model $languages_model Model for the languages.
+	 * @param Options         $options         Polylang's options.
+	 * @param PLL_Cache       $cache           Internal non persistent cache object.
 	 *
-	 * @param PLL_Model $model Instance of `PLL_Model`, passed by reference.
+	 * @phpstan-param PLL_Cache<mixed> $cache
 	 */
-	public function __construct( PLL_Model &$model ) {
-		$this->model          = $model;
-		$this->tax_to_cache[] = $this->tax_language;
+	public function __construct( Languages_Model $languages_model, Options $options, PLL_Cache $cache ) {
```

Comment:
I would not have changed the signature. What advantage to you see?

Reply:
The idea was to prevent a recursion (not storing the model in a property). But now I see that it can be achieved without changing the signature.

Reply:
Reverted the signature in [fe71443](https://github.com/polylang/polylang/pull/1503/commits/fe71443a5a677d64e9262b7a96fb13d46cd2ca4d), for now.